tests/mapper/test_link_mappers.py

```diff
@@ -783,3 +783,390 @@ def test_Attri2VecLinkGenerator_unsupervisedSampler_sample_generation(self):
+
+        assert len(mapper) == 2
+
+        for batch in range(len(mapper)):
+            nf, nl = mapper[batch]
+
+            assert len(nf) == 2 ** (len(self.in_samples) + 2) - 2
+
+            for ii in range(len(nf) // 2):
+                assert nf[2 * ii].shape == nf[2 * ii + 1].shape
+
+            # head nodes
+            for ii in range(2):
+                assert nf[ii].shape == (min(self.batch_size, data_size), 1, self.n_feat)
+
+            # first hop in samples
+            for ii in range(2, 4):
+                assert nf[ii].shape == (
+                    min(self.batch_size, data_size),
+                    self.in_samples[0],
+                    self.n_feat,
+                )
+
+            for ii in range(4, 6):
+                assert nf[ii].shape == (
+                    min(self.batch_size, data_size),
+                    self.out_samples[0],
+                    self.n_feat,
+                )
+
+            for ii in range(6, 8):
+                assert nf[ii].shape == (
+                    min(self.batch_size, data_size),
+                    self.in_samples[0] * self.in_samples[1],
+                    self.n_feat,
+                )
+
+            for ii in range(8, 10):
+                assert nf[ii].shape == (
+                    min(self.batch_size, data_size),
+                    self.in_samples[0] * self.out_samples[1],
+                    self.n_feat,
+                )
+
+            for ii in range(10, 12):
+                assert nf[ii].shape == (
+                    min(self.batch_size, data_size),
+                    self.out_samples[0] * self.in_samples[1],
+                    self.n_feat,
+                )
+
+            for ii in range(12, 14):
+                assert nf[ii].shape == (
+                    min(self.batch_size, data_size),
+                    self.out_samples[0] * self.out_samples[1],
+                    self.n_feat,
+                )
+
+        with pytest.raises(IndexError):
+            nf, nl = mapper[2]
+
+    def test_DirectedGraphSAGELinkGenerator_shuffle(self):
+        def test_edge_consistency(shuffle):
+            G = example_graph(feature_size=1, is_directed=True)
+            edges = list(G.edges())
+            edge_labels = list(range(len(edges)))
+
+            mapper = DirectedGraphSAGELinkGenerator(
+                G, batch_size=2, in_samples=[0], out_samples=[0]
+            ).flow(edges, edge_labels, shuffle=shuffle)
+
+            assert len(mapper) == 2
+
+            for batch in range(len(mapper)):
+                nf, nl = mapper[batch]
+                e1 = edges[nl[0]]
+                e2 = edges[nl[1]]
+                assert nf[0][0, 0, 0] == e1[0]
+                assert nf[1][0, 0, 0] == e1[1]
+                assert nf[0][1, 0, 0] == e2[0]
+                assert nf[1][1, 0, 0] == e2[1]
+
+        test_edge_consistency(True)
+        test_edge_consistency(False)
+
+    def test_DirectedGraphSAGELinkGenerator_not_Stellargraph(self):
```

Comment:
IMO I don't think this test is necessary and could be removed to reduce the amount of code here... or to me it doesn't seem any more valuable than replacing this with:
```python
    def test_not_stellargraph(self):
        with pytest.raises(TypeError):
            DirectedGraphSAGELinkGenerator(None, ...)
```